### DISCUSSION TITLE
Prime One Figma Kit customization

### DISCUSSION THREAD

**Discussion by kedXD** [primefaces/community#3187] [Figma UI Kit]
I purchased the license of Prime One Figma Kit, I need to know is there is any best practice / guidelines to customize the UI kit:

1. I already have a set of colors based on the corporate branding. So, need to use those in place of the ones provided in the Prime One Figma Kit. If I want to replace my colors and shades of blue, red, orange, gray, green etc. what would be the best way to achieve this, knowing that these colors are all mapped to many other components.
2. If I just go ahead and use my own sets of colors and shades, what would be the impact when there will be future release of Prime Figma Kit?
3.  Is 'Token Studio' capable of generating color palette that I can replace with the existing color palettes of blue, green etc? I tried the free version of Token Studio but wasn't able to replace my sets of blue color palette with Prime One blue palette.

Thank you.

**Comment by onursenture** [comment]
The best place to look for Best Practices and Guidelines is the sections on the UI Kit file's Getting Started page. We plan to create short YouTube tutorials on these topics soon. Let me briefly answer your questions:

The best way to change colors for your corporate branding is to use the design tokens we've created in Tokens Studio. By changing the primary tokens under the Semantic Set, you can apply your brand colors wherever they are used. 

<img width="409" alt="image" src="https://github.com/user-attachments/assets/83db2ade-b947-4881-85d8-f4bd7c69a35f" />

If you want to change surface tokens, these are located in the Semantic/Light and Semantic/Dark sets.

<img width="408" alt="image" src="https://github.com/user-attachments/assets/34587b72-b9cb-4abb-aea4-326a788e181c" />

Changes made can be disrupted with each update. The recommended best practice is to make changes not directly on PrimeOne but on a higher-level design system that wraps PrimeOne. This way, you have more control over any breaking changes in PrimeOne and your customizations. Additionally, I recommend documenting everything you do.

Tokens Studio's "Color Scale Generation" is not in the plugin itself but is available in the Graph Engine project, which is still in beta and hasn't been tested by us yet. I can't provide details as I haven't tried it, but you can check it out via this link: [Graph Engine](https://tokens.studio/graph-engine)

**Reply by kedXD** [reply]
I changed the primary color (emerald) to my own set of colors using Token studio, though the changes get reflected in the file the variables remain unchanged. So, my primary color is customized blue now, but the variable palette still shows the default Prime One primary colors. It's very confusing and also the changes take so much of time to reflect, it kind of slows me down every time I make any changes. 
The best practices/guidelines are superficial and hardly helps. More elaborate help system or guides or tutorial videos are a must for these kind of UI kits.
I thought buying a pre-defined UI kit will save time and efforts, but now it takes even more time and lot of effort to customize and on top of it, there is a risk of changes made in local might get broken or disrupted with every upgrade? Thats not good. 
I wish there were more transparent information before one makes a decision to buy the kit.

**Reply by ndupasquier** [reply]
I'm experiencing the same issues, very very long to custom via token studio, change are not reflected in figma variable which is confusing. 
To update variable in figma, I guess you have to click "styles and variables" and export styles and variables but when i do this, it creates like 4000 new variable whereas variable already exist in the figma file. I don't understand why.

**Reply by vla2307** [reply]
Omg, yes, I am facing the same problems. It managed in my test phase to add a color set and put that as primary color, and once the colors even changed in the file and the variables in the file changed too but now when I try it again just a few colors update, so when I put a color by token studio on a rectangle it still shows the old color. I also have problems with exporting the variables since the variable names don't match with variable names of other prime products for example the ULTIMA theme. So I don't see how now the file should help us be faster 😅

**Reply by guacamuller** [reply]
@kedXD @ndupasquier @vla2307 it's been a few months, did you guys find a good workflow to do this? I'm struggling with it myself now. It's super frustrating, easily applying my freaking brand colors should be the most basic thing to do with a UI Kit. Looking for answers mostly just takes me to other people's posts complaining or asking the same things. No clear solutions from the Prime team :(

**Reply by Frederic911** [reply]
@erikvlmnt Thanks so much for raising this question and for taking the time to dig into it. I’m currently considering purchasing the UI kit, and this exact issue is my main concerns—I’m trying to avoid investing extra time into customization or running into limitations that reduce flexibility.

Have you had any updates since your post or found a potential workaround for this problem? I’d really appreciate any insights before making a decision 😊

**Reply by guacamuller** [reply]
Hey @Frederic911! Customizing the Figma kit is very frustrating, but possible. It's also confusing because it changes per version; e.g. Kit v2.2.0 for Angular 14 and Kit 3.1.0 for Angular 19 are set up differently, so a solution that worked in one might not work in the other.

From a designer POV: Most of the frustration comes from the Tokens Studio plugin UX, which is 'mandatory' to use to customize the kit (you can use the free version). Because it's a third-party plugin, Prime kinda differs those issues to them ("go check the Tokens Studio documentation"). The Prime team is helpful here in the forums, but I think that with Tokens Studio being an unavoidable part of the workflow, Prime should properly document and explain customers how to use it with the Kits.

Design tokens make sense from an engineer-y POV, but can feel over-engineered to designers. If you're working with a small team on a simple platform, it might be too much. If you're working in a larger system, it might be a good investment. Make sure to check with your dev team if the handoff flow works for them too (you should check the Theme Designer for this as well, which is another paid Prime product...). 

To me personally, setting up the latest Figma Kit took a lot more time than I though. It's confusing and slow. But I got it to work and it's been smoother sailing since then. It's only been a short amount time, so we'll see how it continues and if new issues pop up. Also, I've only shipped one very simple, self-contained feature using Prime so far, so not a lot of insights yet.
 
Hope this helps!

**Reply by Frederic911** [reply]
Hey @erikvlmnt , wow, I’m genuinely amazed by your response — thank you so much for taking the time to dive deeper into this topic 🙏 Your answer was incredibly helpful, honestly even beyond what I expected. You’re almost doing the work of the PrimeNG support team.

This will definitely help me evaluate our approach in more detail. I hope it’s okay if I reach out again in the future, as the project I’m working on is quite complex.

Thanks again for everything — really appreciate it!

**Reply by guacamuller** [reply]
@Frederic911 Honestly, I'm figuring it out as I go because I pushed for this to be bought so now I kinda have to make it work lol. Not an expert on this in the slightest. But I'm glad my response was helpful and yeah, feel free to reach out. Let me know if you do end up going with this, I might have some questions myself in the future! Good luck with your project!

**Reply by Frederic911** [reply]
@erikvlmnt In my case, I probably will have to work with it anyway since a client of mine used PrimeNG to build their product — double lol. But let's definitely stay in touch. Maybe we can help each other out :)

**Reply by zk-wgo** [reply]
Hi @erikvlmnt, would it be possible for you to share some more details on how to make changes to the PrimeOne UI Kit? I'm (maybe) in a similar position like you were – I pushed for it but didn't expect it to be _SO_ unintuitive! I almost regret my decision to purchase it, but that doesn't help here any more… Would it be ok for you to get in touch with me somehow beyond this forum? Any private messages or maybe (even) a video call?

**Reply by guacamuller** [reply]
Hey @zk-wgo, sorry to hear it happened to you too, it does suck! 
I used version 3.1.0. The hardest thing is understanding the freaking Tokens Studio plugin.

In short, this is the workflow I used to make changes. Not sure if it still works, but you can try:
1. Open the Prime Kit Figma file.
2. Fire up the Tokens Studio plugin - it will be preloaded with all the file's tokens. 
3. There's 3 horizontal tabs at the top: Tokens, Inspect, Code.
4. In **Tokens**, you'll see vertical tabs for categories (Primitive, Semantic, Light, Dark... etc.). Within these categories, you'll find all tokens: colors, spacing, font weights, etc.
5. Understanding which tokens are in which categories is impossible. Good luck. 
6. Find the token(s) you want to edit (e.g. primary color scale is in Semantic/Color/Primary) -> right click -> edit token -> change value -> save. 
7. Once you're done with changes, on the bottom left click "**Styles & Variables**" -> "Export styles & variables to Figma". Yes, even if it says "export", the styles and variables will be updated internally, within this file.
8. You'll be presented with an "Options" screen. Do not change anything and click "Confirm".
9. You'll be presented with an "Export to Figma" screen. It has two tabs in it: "Themes (get pro)" and "Token Sets". 
10. Select '**Themes (get pro)**'. Yes, even if it looks like a pro-only feature, it works for free users.
11. Contain the very intense desire to shout curse words out loud.
12. Reflect on your life's decisions that led you to having to deal with this POS plugin. 
13. Click '**Export**'. 
14. Wait... This will take a while, as it will update thousands of variables within the figma file.
15. After a few minutes, come back and check if the changes have been applied. 

To change fonts to my brand's typeface, I had to use a "font family swapper" plugin (can't remember which one) to bulk replace all fonts in the file. Typography, at least in version 3.1.0, does not use any styles. 

Alternatively, you can disregard all of this and just edit the figma variables directly. This will depend a lot on how dev handoff works in your company or if you're planning to use any of the Primefaces "export to code" products -- otherwise, you may be fine avoiding this and just doing whatever makes more sense to you and other designers. I went through all the work of trying to figure out tokens and how the Prime library is supposed to be used, wrote down a lot of documentation, and then my devs did whatever the hell they wanted and hardcoded the changes their own way. As long as you make sure what you see in figma = kinda what you see in prod, maybe that's good enough.

Hope this helps. Not sure if there's anything else I can do, please remember I have no idea how this works, but you can find my linkedin link in my bio to stay in touch.

**Reply by zk-wgo** [reply]
Hi @erikvlmnt , oh wow, thanks so much for your time and this detailed reply. This confirms my suspicion that this will be a nightmare-like endeavor… I can't image at all why they (the PrimeWhatever guys) set this up like this. The claim "Design To Theme" is close to a lie in my opinion. Yes it's technically feasible but not in a way like they advertised it. Surely won't recommend this product to anyone else.

Anyway, thanks again for your efforts. I'll try to cut my way through this mess as I don't have the possibility to revert the whole thing…

Best regards, wgo

**Reply by steffen-at-visma** [reply]
You guys might find this [PrimeOne guide](https://primevue.org/uikit/guide/) useful. Along with similar instructions as those given by @erikvlmnt, you can also find other useful information.

**Reply by zk-wgo** [reply]
Yes, thanks for the link – of course I have read through all this.

**Reply by guacamuller** [reply]
@steffen-at-visma interesting, that guide was not available when I went through this. Glad they're listening to customer feedback and trying to provide better instructions now, it's a start!

**Reply by steffen-at-visma** [reply]
My team is using PrimeReact, where the guide isn't present on the respective website. I only just found it today — unfortunately _after_ I've trial-and-errored myself to the knowledge myself.

**Comment by ndupasquier** [comment]
> Changes made can be disrupted with each update. The recommended best practice is to make changes not directly on PrimeOne but on a higher-level design system that wraps PrimeOne.

Can you say more about this idea  to wrap primeOne on an higher-leve design system ? PrimeOne is such a huge kit, it's take so much time to make change on colors and apply it. It's hard to browse the file so the idea to wrap PrimeOne can be interesting.

**Reply by onursenture** [reply]
Sorry, we haven't conducted any experiments on this topic. Therefore, any suggestions I provide may not be helpful. A UI Kit of this size can be used in various use cases, making writing general best practices for its advanced and context-specific applications challenging.

**Comment by Animesh-vicinus** [comment]
I changed primary colors of the Figma Kit through "Figma Token" for both ligth and dark versions. Though when I import the component to other file to work upon it, and chnage it to dark won't it would still stay on light mode.

Uploading Screen Recording 2025-04-30 at 10.16.07 AM.mov…

Here you can see.

**Reply by onursenture** [reply]
We couldn't view the screen recording. Could you upload it again so we can review it?

**Reply by Animesh-vicinus** [reply]
I am sorry, the github is not letting me upload the sreen recording

<img width="864" alt="image" src="https://github.com/user-attachments/assets/49c22caa-2152-468b-9db8-23644b603927" />

**Reply by onursenture** [reply]
Could you upload somewhere else, like Dropbox?

**Reply by Animesh-vicinus** [reply]
Hey, I was able to change the colors.

**Reply by earlsinclair-official** [reply]
@Animesh-vicinus can you please explain how you achieved that?
I’m experiencing the same issue. When I change a color token — for example, from blue-500 to pink-500 — the update doesn’t reflect in the Figma design. It only changes when I use “Apply to selection”, but that breaks the connection to the token and applies just the hardcoded hex value instead.

How did you solve this?

**Reply by Animesh-vicinus** [reply]
Hey! I;ll suggets you to maunally set colors through the Variables options given in the Figma, No plugin needed. cause It would take same time to get updated. Also make sure to Publish the assests in Primeone file and update the assests in origin file,
